plugins/reach/lib/kReachManager.php

```diff
@@ -780,60 +772,103 @@ public static function addEntryVendorTask(entry $entry, ReachProfile $reachProfi
+		if ($dictionary)
 		{
-			$status = EntryVendorTaskStatus::PENDING_ENTRY_READY;
+			$entryVendorTask->setDictionary($dictionary->getData());
 		}
-		
-		//Kaltura Recorded entries are ready on creation so make sure the vendors wont fetch the job until it gets its assets
-		if($entry->getSourceType() == EntrySourceType::KALTURA_RECORDED_LIVE && $vendorCatalogItem->requiresEntryReady())
+
+		return $entryVendorTask;
+	}
+
+	protected static function shouldAddEntryVendorTask($entry, $entryObjectType, $vendorCatalogItem)
+	{
+		if($entryObjectType == KalturaEntryObjectType::ENTRY)
 		{
-			$entryAssets = assetPeer::retrieveReadyFlavorsByEntryId($entry->getId());
-			if(!count($entryAssets))
+			//Check if the entry is temporary, if so, dont create the task
+			if($entry->getIsTemporary())
+			{
+				KalturaLog::debug("Entry [{$entry->getId()}] is temporary, entry vendor task object wont be created for it");
+				return false;
+			}
+
+			//Check if static content and the catalog item is excluding static content, if so, dont create the task
+			if(count($vendorCatalogItem->getAdminTagsToExcludeArray()) && array_intersect($vendorCatalogItem->getAdminTagsToExcludeArray(), $entry->getAdminTagsArr()))
+			{
+				KalturaLog::debug("Entry [{$entry->getId()}] has admin tags that are excluded by the catalog item, entry vendor task object wont be created for it");
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	protected static function getEntryVendorTaskStatus($reachProfile, $vendorCatalogItem, $entry, $entryObjectType, $validateModeration)
+	{
+		$status = EntryVendorTaskStatus::PENDING;
+
+		if ($validateModeration && $reachProfile->shouldModerate($vendorCatalogItem->getServiceType()))
+		{
+			$status = EntryVendorTaskStatus::PENDING_MODERATION;
+		}
+
+		if($entryObjectType == KalturaEntryObjectType::ENTRY)
+		{
+			if($vendorCatalogItem->requiresEntryReady() && $entry->getStatus() != entryStatus::READY)
 			{
 				$status = EntryVendorTaskStatus::PENDING_ENTRY_READY;
 			}
+
+			//Kaltura Recorded entries are ready on creation so make sure the vendors wont fetch the job until it gets its assets
+			if($entry->getSourceType() == EntrySourceType::KALTURA_RECORDED_LIVE && $vendorCatalogItem->requiresEntryReady())
+			{
+				$entryAssets = assetPeer::retrieveReadyFlavorsByEntryId($entry->getId());
+				if(!count($entryAssets))
+				{
+					$status = EntryVendorTaskStatus::PENDING_ENTRY_READY;
+				}
+			}
 		}
-		
-		$dictionary = $reachProfile->getDictionaryByLanguage($vendorCatalogItem->getSourceLanguage());
-		if ($dictionary)
-			$entryVendorTask->setDictionary($dictionary->getData());
 
-		$entryVendorTask->setStatus($status);
-		return $entryVendorTask;
+		return $status;
 	}
 	
 	//For automatic dispatched tasks make sure to set the entry creator user as the entry owner
-	protected static function getTaskKuserId(entry $entry)
+	protected static function getTaskKuserId(entry $entry, $entryObjectType)
```

Comment:
We should use different approach, since this function is getting only entry objects as input - there is no use for sending entryObjectType